### ISSUE TITLE
javascript.d.ts exports Blockly.Generator itself instead of its instance

### ISSUE THREAD

**Issue by yukiya383** [google/blockly#4742]
**Describe the bug**
Instead of using Blockly.JavaScript, we can import it from "blockly/javascript" in below statement:
```typescript
import * as JavaScript from "blockly/javascript";
```
Then, we can use some useful functions such as:
```typescript
const value = JavaScript.valueToCode(block, 'value', JavaScript.ORDER_ASSIGNMENT||16)||"";
```
It can be executed without error, but can't pass type check, and error displayed in vscode.

**To Reproduce**
Go to [my repository to reproduce this error](https://github.com/yukiya383/Blockly.JavaScript_bug_minimal_reproduce).
Follow the steps written in README.md.
- Note that the above repository includes other errors which remains after this bug fixed, see its README for more info about them.

**Expected behavior**
JavaScript.valueToCode passes type check.

**Screenshots**
A code using Blockly package:
![JavaScript_is_treated_as_typeof_Generator](https://user-images.githubusercontent.com/49361967/112977360-ec420580-9190-11eb-8ea5-775d562933ee.png)
The same code using [my plugin to fix this bug](https://github.com/yukiya383/blockly-plugin-typed-codegenerator) (expected behavior):
![JavaScript_is_treated_as_Generator](https://user-images.githubusercontent.com/49361967/112977357-eb10d880-9190-11eb-91b5-b7627488cbec.png)
- Note that these are of my local project different from [my repository to reproduce this error](https://github.com/yukiya383/Blockly.JavaScript_bug_minimal_reproduce).

**Stack Traces**

```
src/block.ts:15:28 - error TS2339: Property 'valueToCode' does not exist on type 'typeof Generator'.

15   const value = JavaScript.valueToCode(block, 'value', JavaScript.ORDER_ASSIGNMENT||16)||"";
```

**Estimated cause**
Copied from README of [my repository to reproduce this error](https://github.com/yukiya383/Blockly.JavaScript_bug_minimal_reproduce).
Current contents of typings/javascript.d.ts are shown below:
```typescript
import * as Blockly from './core';
export = Blockly.Generator;
```
This means that, when you import 'blockly/javascript', its type is treated as *typeof* Blockly.Generator.
However, javascript.js exports Blockly.JavaScript, which is **an instance of** Blockly.Generator.

**Suggested fix**
That type declaration should be changed to the one below:
```typescript
import * as Blockly from './core';
declare var tmp:Blockly.Generator;
export = tmp;
```
to export **an instance of** Blockly.Generator instead of *typeof* Blockly.Generator.

**Additional context**
You know, Blockly.JavaScript is not defined in blockly.d.ts, and therefore can't be used in typescript project.
Therefore, it is very preferable if we can directory import it **with proper type**.
If type of JavaScript is properly set to Blockly.Generator, we can easily fix other type problems around it by extending Blockly.Generator properly and downcasting JavaScript to the extended class.
However, if type of JavaScript is **not** properly set, we can't do such hack easily: we must change blockly itself, or cast JavaScript to **any** to solve type errors.
This is why I want this bug fixed.

**Comment by rachel-fenichel** [comment]
Thanks for the detailed repro steps! I've added this to our next release milestone.

**Comment by yukiya383** [comment]
I found that typings/javascript.d.ts was made in PR #3821, which aimed to give type to imports from "blockly/core", and other "blockly/*".
Also, I found similar solution pointed out in #2995, which is overriding module "blockly/javascript" in each project which uses blockly.
It seems to work fine, but fixing javascript.d.ts seems better.

**Comment by yukiya383** [comment]
I made a plugin to get a proper type declaration: https://github.com/yukiya383/blockly-plugin-typed-codegenerator.

**Comment by samelhusseini** [comment]
https://github.com/google/blockly/pull/3821 resolves this issue :)

**Comment by cpcallen** [comment]
@samelhusseini: should this be closed?  This bug [appears to be ABOUT PR #3821](https://github.com/google/blockly/pull/3821#pullrequestreview-624896197).

**Comment by samelhusseini** [comment]
@cpcallen this has been fixed in https://github.com/google/blockly/pull/4753/files
